javacomparator/DiffObject.java

```diff
@@ -0,0 +1,32 @@
+import java.util.ArrayList;
+
+/**
+ * Each of the Diff between some tag is stored as the object of this class.
+ */
+public class DiffObject {
+    String tag;
+    String type;
+    ArrayList<String> content1, content2;
+    String details;
+
+    /**
+     * @param tag    Tag in which this diff was found
+     * @param obj1   The content of the tag in the first file
+     * @param obj2   The content of the tag in the second file
+     * @param detail Some text details for the reason of Diff
+     */
+    public DiffObject(String tag, String type, ArrayList<String> obj1, ArrayList<String> obj2, String detail) {
```

Comment:
Please change the names of the objects (as mentioned in previous comments already) to reflect what they refer to - rather than these names like obj1 and all.